### PR TITLE
feat: Use ternary operator for conditional assignments

### PR DESCRIPTION
**Pull request by webmaster128**
That's the second step initially proposed in https://github.com/stephenh/ts-proto/issues/371. In cases where the input value needs to be transformed, we [cannot use nullish coalescing](https://stackoverflow.com/questions/69696665/optional-chaining-with-free-functins) or anything new and fancy.

This improves the code output in two ways:
- Less code (that ultimatly reduces the app bundle size for the end user)
- Exactly one assignment per field which I find easier to reason about

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1212,21 +1212,17 @@ function generateFromPartial(ctx: Context, fullName: string, messageDesc: Descri
       const fallback = isWithinOneOf(field) ? 'undefined' : defaultValue(ctx, field);
       chunks.push(code`message.${fieldName} = ${fallback}`);
       chunks.push(code`}`);
-    } else if (
-      isPrimitive(field) ||
-      (isTimestamp(field) && (options.useDate === DateOption.DATE || options.useDate === DateOption.STRING)) ||
-      isValueType(ctx, field)
-    ) {
+    } else if (readSnippet(`x`).toCodeString() == 'x') {
```
Comment (webmaster128) [inline src/main.ts:1215]: This somehwat unrelated change is worth having a 👀.

Comment (stephenh) [reply src/main.ts:1215]: Ah yeah, that's a neat way of doing this.

**Review by stephenh — APPROVED** [review]
Looks great!

**Comment by webmaster128** [comment]
Thanks!
